docs/features/opencdc-record.mdx

```diff
@@ -56,9 +80,84 @@ When processing records in Conduit, you can always expect a similar structure to
 ```
 
 :::note
-`.Position`, `.Key`, and `.Payload.Before` are represented as `Base64` encoded in the example above because these will be a byte slice when represented as JSON.
+`.Position`, `.Key`, and `.Payload.Before` are represented as `Base64` encoded
+in the example above because these will be a byte slice when represented as
+JSON.
 :::
 
+## Data types
+
+
+There are no limitations when it comes to data types a source connector can read
+from a source. Certain type limitations might apply depending on how the data is
+moved internally in Conduit, from source connector to Conduit and from Conduit
+to destination connectors.
+
+The key and the payload data can be represented in two ways: raw and structured.
+
+## Raw data
+
+Raw data in Conduit is an array of bytes that can represent the source data in
+its original form (such as a file) or encoded in a certain way (for example, a
+JSON representation of a database row or an Avro encoded structure). It's
+represented by the `opencdc.RawData` Go type. For example, when writing a
+connector or a processor in Go, you can construct raw data like this:
+
+```go
+opencdc.RawData([]byte{1, 3, 5})
+```
+
+## Structured data
+
+Structured data in Conduit is a map in which the keys are field names and values
+are field values. An example of that is the below record's `.Payload.After`
+field: 
+
+
+```json5
+{
+    // other record fields
+    "payload": {
+        "before": "eWVsbG93",
+        "after": {
+            "bool_field": true,
+            "float_field": 1.2,
+            "int_field": 1,
+            "string_field": "orange"
+        }
+    }
+}
+```
+
+When writing a connector or a processor in Go, it's represented by
+the `opencdc.StructuredData` type. 
+
+The supported data types for values in `opencdc.StructuredData` depend on following:
+- connector or processor type (built-in or standalone)
+- [schema support](/docs/features/schema-support) (enabled or disabled).
+
+In built-in connectors, the field values can be of any Go type, given that
+there's no (de)serialization involved.
+
+In standalone connectors with schema middleware enabled (which is the default),
+any type that is supported by the [Apache Avro™ format](https://avro.apache.org)
+is also supported by Conduit.
+
+If the schema middleware is disabled in a connector, then the supported types
+are limited to what Protobuf allows as
+a [value](https://protobuf.dev/reference/protobuf/google.protobuf/#value). That
+translates to the following Go types:
+
+* `bool`
+* `int`, `int32`, `int64`, `uint`, `uint32`, `uint64`
+* `float32`, `float64`
+* `string`
+* `[]byte` (stored as a string, base64-encoded)
+* `map[string]interface{}` (a map of strings to any of the values that are supported)
+* `[]interface{}` (a slice of any value that is supported)
+
+A notable limitation is timestamps, i.e. `time.Time` values are not supported.
```

Comment:
They _are_ supported now that we have schema support!

Reply:
I realized these listed types explain what's supported if schemas are disabled. This is fine.